pkg/services/ngalert/notifier/receiver_svc.go

```diff
@@ -567,3 +567,47 @@ func makeErrReceiverVersionConflict(current *models.Receiver, desiredVersion str
 	}
 	return ErrReceiverVersionConflict.Build(data)
 }
+
+func makeErrReceiverDependentResourcesProvenance(usedByRoutes bool, rules []models.AlertRuleKey) error {
+	uids := make([]string, 0, len(rules))
+	for _, key := range rules {
+		uids = append(uids, key.UID)
+	}
+	data := make(map[string]any, 2)
+	if len(uids) > 0 {
+		data["UsedByRules"] = uids
+	}
+	if usedByRoutes {
+		data["UsedByRoutes"] = true
+	}
+
+	return ErrReceiverDependentResourcesProvenance.Build(errutil.TemplateData{
+		Public: data,
+	})
+}
+
+func (rs *ReceiverService) RenameReceiverInDependentResources(ctx context.Context, orgID int64, route *definitions.Route, oldName, newName string, receiverProvenance models.Provenance) error {
```

Comment:
make it public to let provisioning service use it without having to call update receiver.